src/include/duckdb/main/config.hpp

```diff
@@ -357,6 +358,12 @@ struct DBConfig {
 	DUCKDB_API optional_ptr<CompressionFunction> GetCompressionFunction(CompressionType type,
 	                                                                    const PhysicalType physical_type);
 
+	//! Returns the decode function matching the encoding name.
+	DUCKDB_API optional_ptr<DecodingFunction> GetDecodeFunction(string name) const;
+	DUCKDB_API void RegisterDecodeFunction(const DecodingFunction &function) const;
+
+	//! Returns the decode function names.
+	DUCKDB_API vector<string> GetLoadedDecodeFunctionNames() const;
```

Comment:
Can we make this return the actual DecodingFunctions instead of only the names (`vector<reference<DecodingFunction>>`)?